src/main/java/com/shapesecurity/shift/es2017/parser/GenericParser.java

```diff
@@ -414,15 +420,17 @@ protected BindingIdentifier parseBindingIdentifier() throws JsError {
     @Nonnull
     protected String parseIdentifier() throws JsError {
         if (this.matchIdentifier()) {
-            if (this.match(TokenType.YIELD) && this.allowYieldExpression) {
+			String value = this.lookahead.toString();
+			if (this.match(TokenType.YIELD) && this.allowYieldExpression || this.allowYieldExpression && this.match(TokenType.ESCAPED_KEYWORD) && value.equals("yield") && ((IdentifierToken) this.lookahead).escaped) {
                 throw this.createError(ErrorMessages.INVALID_TOKEN_CONTEXT, "yield");
-            } else if (this.match(TokenType.AWAIT) && (this.allowAwaitExpression || this.moduleIsTheGoalSymbol)) {
+            } else if ((this.allowAwaitExpression || this.moduleIsTheGoalSymbol) && (this.match(TokenType.AWAIT) || this.match(TokenType.ESCAPED_KEYWORD) && value.equals("await") && ((IdentifierToken) this.lookahead).escaped )) {
```

Comment:
Why is the last condition necessary? Is it ever false, given the earlier `this.match(TokenType.ESCAPED_KEYWORD)`?